### ISSUE TITLE
Different PATH value in action and EC2

### ISSUE THREAD

**Issue by ay0o** [philips-labs/terraform-aws-github-runner#2943]
I have an outdated version (0.18.1) of this module running in production.

In order to upgrade to 2.1.1, I'm deploying a new instance of the module. So far so good, the infra is deployed, the webhook creates EC2, the EC2 are registered as runners in GH and the actions are scheduled in the new runners.

The issue I'm facing is that the PATH is differing. 

I'm using the same AMI for the old and the new infrastructure. Some tools like `aws`, `docker`, etc are installed in this AMI, and the binaries are available at `/usr/local/bin`.

Now, here starts the strange part. Let's use this code as sample:

```
name: Test new runner

on:
  workflow_dispatch:

jobs:
  test:
    runs-on: new
    steps:
      - run: echo $PATH
      - run: aws help
```

When it runs in the old runner, everything is fine. However, when it runs in the new, it crashes because it can't find `aws`.

Checking the output of PATH, that's because `/usr/local/bin` is missing:
```
Run echo $PATH
/sbin:/bin:/usr/sbin:/usr/bin
```

However, if I log into the runner and run `echo $PATH`, the output is `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin`. The path is included. The action is running as `root` as well, but just for testing purposes, I checked the path of the other users `ec2-user` and `ssm-user`, and `/usr/local/bin` is also available in both.

Any ideas about why this might be happening? I can workaround this by building a new AMI with the tools installed to `/usr/bin`, but I'm trying to avoid this, as `/usr/local/bin` should be a well-known path to be added in `PATH`.

**Comment by npalm** [comment]
As part of the upgrade from 1.x to 2.x the SSM paramaters are changed. Which also contains the run_as. Are you sure you updated runner still running as root? By default it tries to run as ec2-user.

**Comment by ay0o** [comment]
yes, `runner_as_root` is set to `true`.

the issue is not the user the runner is using, but the PATH is actually different for the same user. And as I said, just for testing purposes I tested all other users, `ec2-user` and `ssm-user` in my case, and with both the same happened.

If I SSH/SSM into the instance, PATH includes `/usr/local/bin`. However, when it's the action the one connecting to the instance, the PATH does not include `/usr/local/bin`.

You might be wondering that maybe there is a misconfiguration somewhere, maybe a different `.bashrc` or `.bash_profile` to name some examples. However, as I said, it's the exact same AMI I'm using with an older version of this working which is working without issues.

For now, I workarounded this by building a new AMI with the binary of the tools located at `/usr/bin`, but still I find it quite odd that the PATH is differing.
